src/main/java/graphql/normalized/NormalizedInputValue.java

```diff
@@ -0,0 +1,61 @@
+package graphql.normalized;
+
+import java.util.Objects;
+
+/**
+ * A value with type information.
+ */
+public class NormalizedInputValue {
+    private final String type;
+    private final Object value;
+
+    public NormalizedInputValue(String type, Object value) {
+        this.type = type;
+        this.value = value;
+    }
+
+    /**
+     * This can be a wrapped type: e.g. [String!]!
+     *
+     * @return
+     */
+    public String getType() {
```

Comment:
Do we have a way to undo this easily?

Imagine

`GraphqlType t = GraphTypeUtil.getTypeFromPrintedString(schema, "[String!]!")`

eg my question is - how does one move from logical name to actual type?

Reply:
good point .. I don't think we have

Reply:
I think it should be a `TypeName` either way, because it encapsulates the same info but in a type safe manner instead of putting it in String notation

Reply:
We already hav a TypeName class which is an AST class which we don't use on purpose.

Reply:
But I agree it should be names `typeName`, not `type`.